src/libraries/System.Text.Json/Common/JsonHelpers.cs

```diff
@@ -1,32 +1,46 @@
 // Licensed to the .NET Foundation under one or more agreements.
 // The .NET Foundation licenses this file to you under the MIT license.
 
-using System.Runtime.InteropServices;
 using System.Collections.Generic;
 using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+using System.Runtime.InteropServices;
 
 namespace System.Text.Json
 {
     internal static partial class JsonHelpers
     {
+#if !NETCOREAPP
         /// <summary>
-        /// Emulates Dictionary.TryAdd on netstandard.
+        /// netstandard/netfx polyfill for Dictionary.TryAdd
         /// </summary>
-        public static bool TryAdd<TKey, TValue>(this Dictionary<TKey, TValue> dictionary, in TKey key, in TValue value) where TKey : notnull
+        public static bool TryAdd<TKey, TValue>(this Dictionary<TKey, TValue> dictionary, TKey key, TValue value) where TKey : notnull
         {
-#if NETSTANDARD2_0 || NETFRAMEWORK
             if (!dictionary.ContainsKey(key))
             {
                 dictionary[key] = value;
                 return true;
             }
 
             return false;
-#else
-            return dictionary.TryAdd(key, value);
-#endif
         }
 
+        /// <summary>
+        /// netstandard/netfx polyfill for Queue.TryDequeue
+        /// </summary>
+        public static bool TryDequeue<T>(this Queue<T> queue, [NotNullWhen(true)] out T? result)
```

Comment:
>TryDequeue [](http://example.com/codeflow?start=27&length=10)

are we adding this because TryDequeue is not supported in ns2.0? should we ifdef it?

Reply:
It is ifdef'd together with the method above.